### DISCUSSION TITLE
draw.io in visual editor

### DISCUSSION THREAD

**Discussion by Kris33** [requarks/wiki#4860] [Help / Questions] [answered]
Hi, 
First things first, great product.

Only one question: I can't find how to use Draw.io from the visual editor.
for me, it's only avaliable from the markdown editor. is this normal?
if so, is there a way to have it integrated with the visual editor?

**Comment by NGPixel** [comment, marked as answer]
It's only available in the markdown editor. It's planned to be added to the visual editor in 3.x.

**Reply by lublak** [reply]
drawio also supports direct integration:

![grafik](https://user-images.githubusercontent.com/44057030/166699607-4423024f-4e23-4852-b0f4-de5ac5757075.png)

https://github.com/jgraph/drawio-integration
https://github.com/jgraph/drawio-integration/blob/master/inline.js
http://jgraph.github.io/drawio-integration/inline.html

**Comment by nhanchaukp** [comment]
I've rebuilt ckeditor5 and integrated a diagram; you can refer to it here: https://github.com/nhanchaukp/wikijs-ckeditor5
